src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -1028,46 +1018,36 @@ private async Task<SearchResult> SearchForReindexSurrogateIdsWithoutSearchParamH
             long endResourceSurrogateId = 0;
 
             SearchResult searchResult = null;
-            await _sqlRetryService.ExecuteSql(
-                async (cancellationToken, sqlException) =>
-                {
-                    using SqlConnection connection = await _sqlConnectionBuilder.GetSqlConnectionAsync(initialCatalog: null, cancellationToken: cancellationToken).ConfigureAwait(false);
-                    using SqlCommand sqlCommand = connection.CreateCommand();
-                    connection.RetryLogicProvider = null; // To remove this line _sqlConnectionBuilder in healthcare-shared-components must be modified.
-                    await connection.OpenAsync(cancellationToken);
 
-                    sqlCommand.CommandTimeout = Math.Max((int)_sqlServerDataStoreConfiguration.CommandTimeout.TotalSeconds, 180);
-
-                    sqlCommand.Parameters.AddWithValue("@p0", resourceTypeId);
-                    sqlCommand.CommandText = @"
+            using var sqlCommand = new SqlCommand();
+            sqlCommand.CommandTimeout = Math.Max((int)_sqlServerDataStoreConfiguration.CommandTimeout.TotalSeconds, 180);
```

Comment:
>sqlCommand.CommandTimeout = Math.Max((int)_sqlServerDataStoreConfiguration.CommandTimeout.TotalSeconds, 180); [](http://example.com/codeflow?start=12&length=109)

This query scans entire resource type. Are you sure that it does not fail for large database?

Reply:
This has been revised per your comment.